File: backend/EmployeeManagement.Api/Models/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace EmployeeManagement.Api.Models;

[Index(nameof(Email), IsUnique = true)]
public class Employee
{
    public int Id { get; set; }

    [Required]
    [StringLength(50)]
    public string FirstName { get; set; } = string.Empty;

    [Required]
    [StringLength(50)]
    public string LastName { get; set; } = string.Empty;

    [Required]
    [EmailAddress]
    [StringLength(150)]
    public string Email { get; set; } = string.Empty;

    [Required]
    [StringLength(100)]
    public string Department { get; set; } = string.Empty;

    [Required]
    [StringLength(100)]
    public string Position { get; set; } = string.Empty;

    [Range(0, double.MaxValue)]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Salary { get; set; }

    public DateTime HireDate { get; set; }

    public bool IsActive { get; set; }
}

File: frontend/src/app/models/employee.ts
export interface Employee {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    department: string;
    position: string;
    salary: number;
    hireDate: string;
    isActive: boolean;
  }
  
  export interface CreateEmployee {
    firstName: string;
    lastName: string;
    email: string;
    department: string;
    position: string;
    salary: number;
    hireDate: string;
    isActive: boolean;
  }
  
  export interface UpdateEmployee {
    firstName: string;
    lastName: string;
    email: string;
    department: string;
    position: string;
    salary: number;
    hireDate: string;
    isActive: boolean;
  }
  
  export interface PagedResult<T> {
    items: T[];
    totalCount: number;
    page: number;
    pageSize: number;
    totalPages: number;
  }